watcher/local/watcher_test.go

```diff
@@ -80,6 +84,107 @@ func Test_StartWatching(t *testing.T) {
+			params, txs := randomTxsForSingleCh(rng, 3)
+			adjSub, trigger := setupAdjudicatorSub(makeRegisteredEvent(txs[2])...)
+
+			rs := &mocks.RegisterSubscriber{}
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSub, nil)
+			w := newWatcher(t, rs)
+			// Start watching and publish states.
+			statesPub, eventsForClient := startWatchingForLedgerChannel(t, w, makeSignedStateWDummySigs(params, txs[0].State))
+			require.NoError(t, statesPub.Publish(txs[1]))
+			require.NoError(t, statesPub.Publish(txs[2]))
+
+			// Trigger events and assert.
+			triggerAdjEventAndExpectNotif(t, trigger, eventsForClient)
+			rs.AssertExpectations(t)
+		})
+		t.Run("happy/newer_than_latest_state_registered", func(t *testing.T) {
+			params, txs := randomTxsForSingleCh(rng, 3)
+			adjSub, trigger := setupAdjudicatorSub(makeRegisteredEvent(txs[2])...)
+
+			rs := &mocks.RegisterSubscriber{}
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSub, nil)
+			w := newWatcher(t, rs)
+			// Start watching and publish states.
+			statesPub, eventsForClient := startWatchingForLedgerChannel(t, w, makeSignedStateWDummySigs(params, txs[0].State))
+			require.NoError(t, statesPub.Publish(txs[1]))
+
+			// Trigger events.
+			triggerAdjEventAndExpectNotif(t, trigger, eventsForClient)
+
+			rs.AssertExpectations(t)
+		})
+	})
+
+	t.Run("ledger_channel_with_sub_channel", func(t *testing.T) {
+		t.Run("happy/latest_state_registered", func(t *testing.T) {
+			parentParams, parentTxs := randomTxsForSingleCh(rng, 3)
+			childParams, childTxs := randomTxsForSingleCh(rng, 3)
+			parentTxs[2].Allocation.Locked = []channel.SubAlloc{{ID: childTxs[0].ID}} // Add sub-channel to allocation.
+
+			adjSubParent, triggerParent := setupAdjudicatorSub(makeRegisteredEvent(parentTxs[2])...)
+			adjSubChild, triggerChild := setupAdjudicatorSub(makeRegisteredEvent(childTxs[2])...)
+
+			rs := &mocks.RegisterSubscriber{}
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSubParent, nil).Once()
+			rs.On("Subscribe", mock.Anything, mock.Anything).Return(adjSubChild, nil).Once()
+
+			w := newWatcher(t, rs)
+			// Parent: Start watching and publish states.
+			parentSignedState := makeSignedStateWDummySigs(parentParams, parentTxs[0].State)
+			statesPubParent, eventsForClientParent := startWatchingForLedgerChannel(t, w, parentSignedState)
+			require.NoError(t, statesPubParent.Publish(parentTxs[1]))
+			require.NoError(t, statesPubParent.Publish(parentTxs[2]))
+
+			// Child: Start watching and publish states.
+			childSignedState := makeSignedStateWDummySigs(childParams, childTxs[0].State)
+			statesPubChild, eventsForClientChild := startWatchingForSubChannel(t, w, childSignedState, parentTxs[0].State.ID)
+			require.NoError(t, statesPubChild.Publish(childTxs[1]))
+			require.NoError(t, statesPubChild.Publish(childTxs[2]))
+
+			// Parent, Child: Trigger events.
+			triggerAdjEventAndExpectNotif(t, triggerParent, eventsForClientParent)
```

Comment:
Maybe you can name this like `assertNEvents` and pass an integer `n` in there for how many events it should await.

Reply:
In both these calls, the arguments are different. In first call, the trigger and eventsSub correspond to ledger channel, while in the second call, they correspond to sub-channel.

So, this will anyway require two calls.